From 25112a8d44e6a0df8eb4fd6e716496229b1d9f31 Mon Sep 17 00:00:00 2001
From: Yipeng Zou <zouyipeng@huawei.com>
Date: Tue, 18 Apr 2023 15:21:56 +0000
Subject: [PATCH 0092/1646] [Huawei] Revert "[Huawei] irqchip/gic-v3-its:
 introduce CONFIG_GENERIC_PENDING_IRQ"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I6WPFT

--------------------------------

This reverts commit 7e2085dadf4255b17fea64c0e98069a62e43bb6c.

Signed-off-by: Yipeng Zou <zouyipeng@huawei.com>
---
 drivers/irqchip/irq-gic-v3-its.c | 8 +-------
 kernel/irq/Kconfig               | 4 +---
 kernel/irq/migration.c           | 2 --
 3 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index 64cfa5e38d89..1cb392fb16d0 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -1678,12 +1678,6 @@ static int its_select_cpu_other(const struct cpumask *mask_val)
 }
 #endif
 
-static void its_irq_chip_eoi(struct irq_data *d)
-{
-	irq_move_irq(d);
-	irq_chip_eoi_parent(d);
-}
-
 static int its_set_affinity(struct irq_data *d, const struct cpumask *mask_val,
 			    bool force)
 {
@@ -2032,7 +2026,7 @@ static struct irq_chip its_irq_chip = {
 	.name			= "ITS",
 	.irq_mask		= its_mask_irq,
 	.irq_unmask		= its_unmask_irq,
-	.irq_eoi		= its_irq_chip_eoi,
+	.irq_eoi		= irq_chip_eoi_parent,
 	.irq_set_affinity	= its_set_affinity,
 	.irq_compose_msi_msg	= its_irq_compose_msi_msg,
 	.irq_set_irqchip_state	= its_irq_set_irqchip_state,
diff --git a/kernel/irq/Kconfig b/kernel/irq/Kconfig
index 8dbd236a4f19..cdf9fbe47026 100644
--- a/kernel/irq/Kconfig
+++ b/kernel/irq/Kconfig
@@ -33,9 +33,7 @@ config GENERIC_IRQ_LEGACY_ALLOC_HWIRQ
 
 # Support for delayed migration from interrupt context
 config GENERIC_PENDING_IRQ
-	bool "Support for delayed migration from interrupt context"
-	depends on SMP
-	default n
+	bool
 
 # Support for generic irq migrating off cpu before the cpu is offline.
 config GENERIC_IRQ_MIGRATION
diff --git a/kernel/irq/migration.c b/kernel/irq/migration.c
index f752d9ff8ee1..1ca2fbf5c32d 100644
--- a/kernel/irq/migration.c
+++ b/kernel/irq/migration.c
@@ -122,5 +122,3 @@ void __irq_move_irq(struct irq_data *idata)
 	if (!masked)
 		idata->chip->irq_unmask(idata);
 }
-
-void __weak irq_force_complete_move(struct irq_desc *desc) { }
-- 
2.17.1

